src/components/implementation/no_interface/cFE_booter/osfiles.c

```diff
@@ -17,201 +23,260 @@ int32 OS_FS_Init(void)
+
+    return ret;
 }
 
 /*
  * Copies the info of an open file to the structure
 */
 int32 OS_FDGetInfo(int32 filedes, OS_FDTableEntry *fd_prop)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if (!fd_prop)                                   return OS_FS_ERR_INVALID_POINTER;
+    if (filedes <= 0 || filedes > MAX_NUM_FILES)    return OS_FS_ERR_INVALID_FD;
+
+    return file_FDGetInfo(filedes, fd_prop);
 }
 
 /*
 ** Check to see if a file is open
 */
 int32 OS_FileOpenCheck(char *Filename)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    int32 ret = path_chk_exists(Filename);
+    if (ret != OS_FS_SUCCESS) return OS_FS_ERR_INVALID_POINTER;
+
+    struct fsobj *file = file_find(Filename);
+    if (!file) return OS_INVALID_POINTER;
+    if (file->refcnt == 0) return OS_FS_ERROR;
+    return OS_FS_SUCCESS;
+
 }
 
 /*
 ** Close all open files
 */
 int32 OS_CloseAllFiles(void)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    uint32 i;
+    for (i = 1 ; i < MAX_NUM_FILES + 1 ; i++) {
+        OS_close(i);
+    }
+    return OS_FS_SUCCESS;
 }
 
 /*
 ** Close a file by filename
 */
 int32 OS_CloseFileByName(char *Filename)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    int32 ret = path_chk_exists(Filename);
+    if (ret != OS_FS_SUCCESS) return ret;
+    return file_close_by_name(Filename);
 }
 
-
 /******************************************************************************
 ** Directory API
 ******************************************************************************/
 
 /*
  * Makes a new directory
+ * access is not used by cFE
 */
 int32 OS_mkdir(const char *path, uint32 access)
 {
-    PANIC("Read Only Filesystem!");
-    return 0;
+    int32 ret = path_chk_isvalid(path);
+    if (ret != OS_FS_SUCCESS) return ret;
+    return file_mkdir((char *)path);
 }
 
 /*
  * Opens a directory for searching
 */
 os_dirp_t OS_opendir(const char *path)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if(path_chk_exists(path) != OS_FS_SUCCESS) return NULL;
+    int32 FD = dir_open((char *)path);
+    if (FD == 0) return NULL;
+    return (os_dirp_t) FD;
 }
 
 /*
  * Closes an open directory
 */
 int32 OS_closedir(os_dirp_t directory)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if (!directory) return OS_FS_ERR_INVALID_POINTER;
+    return dir_close((int32) directory);
 }
 
 /*
  * Rewinds an open directory
 */
 void OS_rewinddir(os_dirp_t directory)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
+    if (!directory) return;
+    dir_rewind((int32) directory);
+    return;
 }
 
 /*
  * Reads the next object in the directory
 */
-os_dirent_t * OS_readdir(os_dirp_t directory)
+os_dirent_t *OS_readdir(os_dirp_t directory)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    if (!directory) return NULL;
```

Comment:
It does and I am removing this line for now.  But I would like to talk further on this.  the API spec says return null if directory is null.  It seems more maintainable if I check for what the api considers invalid inputs at this level, and then check for what the underlying file system considers a valid input at the filesystem level.  That way changes can be made to the filesystem without breaking the API level.  Thoughts?